### PR TITLE
[Browserview] 4.6 and 4.7 PRs

### PR DESCRIPTION
**Pull request by Willyfrog**
**Summary**
adds a commit per PR so it can be easily compared.
it is not a straight cherry pick, as many components have moved and some
**Issue link**
[v4.6](https://github.com/mattermost/desktop/pulls?q=is%3Apr+is%3Aclosed+milestone%3Av4.6.0)
[v4.7](https://github.com/mattermost/desktop/pulls?q=is%3Apr+is%3Aclosed+milestone%3Av4.7.0)

**Additional Notes**
[Dean's new loading screen PR](https://github.com/mattermost/desktop/pull/1409) is missing due to breaking after a few seconds due to memory consuption (most likely I introduced a bug while translating, but I need to investigate)

### CODE REVIEW

File: src/main/main.js [hunk 1]
```diff
@@ -887,16 +889,25 @@ function handleUpdateMenuEvent(event, configData) {
   }
 }
 
+async function handleSelectDownload(event, startFrom) {
+  const message = 'Specify the folder where files will download';
+  const result = await dialog.showOpenDialog({defaultPath: startFrom,
+    message,
```
Comment (nevyangelova) [inline src/main/main.js:895]: Nit: do you have eslint enabled? I feel like in a few places ident and spacing is not matching the settings in the webapp, maybe I am not seeing correctly tho :D

File: src/main/notifications/Mention.js [hunk 2]
```diff
@@ -34,7 +33,7 @@ export class Mention extends Notification {
     this.customSound = customSound;
   }
 
-  onShow = () => {
-    playSound(this.customSound);
+  getNotificationSound = () => {
+    return this.customSound;
   }
 }
```
Comment (nevyangelova) [inline src/main/notifications/Mention.js:39]: nit: as well as the empty lines

File: src/main/preload/mattermost.js [hunk 3]
```diff
@@ -72,8 +72,6 @@ document.addEventListener('mouseup', () => {
 
 // listen for messages from the webapp
 window.addEventListener('message', ({origin, data: {type, message = {}} = {}} = {}) => {
-  console.log(`got a message from ${origin} of type ${type}`);
-  console.log(message);
   if (origin !== window.location.origin) {
     return;
```
Comment (nevyangelova) [inline src/main/preload/mattermost.js:76]: nit: here as well. Is it possible to do an autofix for these files once they are merged?

File: src/main/viewManager.js [hunk 4]
```diff
@@ -63,9 +63,13 @@ export class ViewManager {
   }
 
   showInitial = () => {
-    // TODO: handle deeplink url
-    const element = this.configServers.find((e) => e.order === 0);
-    this.showByName(element.name);
+    console.log('configservers');
+    console.log(this.configServers);
```
Comment (nevyangelova) [inline src/main/viewManager.js:67]: nit: console.log

**Review by nevyangelova — APPROVED** [review]

File: src/main/main.js (hunk 1 above)
Comment (devinbinnie) [reply src/main/main.js:895]: Indenting in the desktop app has always been 2 spaces for some reason. We should fix that at some point.

**Review by devinbinnie — APPROVED** [review]

**Comment by Willyfrog** [comment]
@nevyangelova added a ticket to fix it: https://mattermost.atlassian.net/browse/MM-31399

**Pull request merged**